### ISSUE TITLE
MSI PRO Z690-A aspm not working

### ISSUE THREAD

**Issue by c67871f2** [Dasharo/dasharo-issues#553]
### Device

MSI PRO Z690-A DDR5

### Dasharo version

4.21

### Brief summary

100%

### How to reproduce

1. poweroff --halt
2. Booting linux with M2_1 with aspm turned on,hangs at boot UefiPayload.

### Expected behavior

Successfully booting linux

### Actual behavior

hangs at boot linux

### Solutions you've tried

I tried this patch to fix the problem. GPP_D0 to none fix pcie_e1 on aspm not working.
```diff --git a/src/mainboard/msi/ms7d25/gpio.h b/src/mainboard/msi/ms7d25/gpio.h
index 210c43d398..8f512ef9d4 100644
--- a/src/mainboard/msi/ms7d25/gpio.h
+++ b/src/mainboard/msi/ms7d25/gpio.h
@@ -166,14 +166,14 @@ static const struct pad_config gpio_table[] = {
 
        /* ------- GPIO Group vGPIO_0 ------- */
        /* These are Virtual USB OC pins */
-       _PAD_CFG_STRUCT(VGPIO_USB_0,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0), /* VGPIO_USB_0 */
-       _PAD_CFG_STRUCT(VGPIO_USB_1,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0), /* VGPIO_USB_1 */
-       _PAD_CFG_STRUCT(VGPIO_USB_2,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0), /* VGPIO_USB_2 */
-       _PAD_CFG_STRUCT(VGPIO_USB_3,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0), /* VGPIO_USB_3 */
-       _PAD_CFG_STRUCT(VGPIO_USB_8,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0), /* VGPIO_USB_8 */
-       _PAD_CFG_STRUCT(VGPIO_USB_9,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0), /* VGPIO_USB_9 */
-       _PAD_CFG_STRUCT(VGPIO_USB_10, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0), /* VGPIO_USB_10 */
-       _PAD_CFG_STRUCT(VGPIO_USB_11, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0), /* VGPIO_USB_11 */
+       PAD_CFG_NF_VWEN(VGPIO_USB_0, NONE, DEEP, NF1), /* VGPIO_USB_0 */
+       PAD_CFG_NF_VWEN(VGPIO_USB_1, NONE, DEEP, NF1), /* VGPIO_USB_1 */
+       PAD_CFG_NF_VWEN(VGPIO_USB_2, NONE, DEEP, NF1), /* VGPIO_USB_2 */
+       PAD_CFG_NF_VWEN(VGPIO_USB_3, NONE, DEEP, NF1), /* VGPIO_USB_3 */
+       PAD_CFG_NF_VWEN(VGPIO_USB_8, NONE, DEEP, NF1), /* VGPIO_USB_8 */
+       PAD_CFG_NF_VWEN(VGPIO_USB_9, NONE, DEEP, NF1), /* VGPIO_USB_9 */
+       PAD_CFG_NF_VWEN(VGPIO_USB_10, NONE, DEEP, NF1), /* VGPIO_USB_10 */
+       PAD_CFG_NF_VWEN(VGPIO_USB_11, NONE, DEEP, NF1), /* VGPIO_USB_11 */
 
        /* ------- GPIO Community 1 ------- */
 
@@ -416,89 +416,89 @@ static const struct pad_config gpio_table[] = {
        PAD_CFG_GPI_TRIG_OWN(GPP_C23, NONE, PLTRST, OFF, ACPI),
 
        /* CPU PCIe CLKREQ virtual wire message buses */
-       _PAD_CFG_STRUCT(VGPIO_PCIE_0,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_1,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_2,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_3,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_4,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_5,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_6,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_7,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_8,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_9,  PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_10, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_11, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_12, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_13, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_14, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_15, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_64, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_65, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_66, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_67, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-
-       _PAD_CFG_STRUCT(VGPIO_PCIE_16, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_17, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_18, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_19, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_20, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_21, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_22, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_23, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_24, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_25, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_26, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_27, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_28, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_29, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_30, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_31, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_68, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_69, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_70, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_71, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-
-       _PAD_CFG_STRUCT(VGPIO_PCIE_32, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_33, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_34, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_35, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_36, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_37, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_38, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_39, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_40, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_41, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_42, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_43, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_44, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_45, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_46, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_47, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_72, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_73, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_74, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_75, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-
-       _PAD_CFG_STRUCT(VGPIO_PCIE_48, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_49, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_50, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_51, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_52, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_53, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_54, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_55, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_56, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_57, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_58, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_59, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_60, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_61, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_62, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_63, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_76, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_77, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_78, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
-       _PAD_CFG_STRUCT(VGPIO_PCIE_79, PAD_FUNC(NF1) | PAD_RESET(DEEP) | PAD_CFG0_NAFVWE_ENABLE, 0),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_0, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_1, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_2, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_3, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_4, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_5, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_6, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_7, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_8, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_9, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_10, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_11, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_12, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_13, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_14, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_15, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_64, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_65, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_66, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_67, NONE, PLTRST, NF1),
+
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_16, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_17, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_18, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_19, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_20, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_21, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_22, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_23, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_24, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_25, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_26, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_27, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_28, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_29, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_30, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_31, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_68, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_69, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_70, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_71, NONE, PLTRST, NF1),
+
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_32, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_33, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_34, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_35, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_36, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_37, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_38, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_39, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_40, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_41, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_42, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_43, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_44, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_45, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_46, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_47, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_72, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_73, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_74, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_75, NONE, PLTRST, NF1),
+
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_48, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_49, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_50, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_51, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_52, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_53, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_54, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_55, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_56, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_57, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_58, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_59, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_60, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_61, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_62, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_63, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_76, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_77, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_78, NONE, PLTRST, NF1),
+       PAD_CFG_NF_VWEN(VGPIO_PCIE_79, NONE, PLTRST, NF1),
 
        /* ------- GPIO Community 4 ------- */
 
@@ -651,7 +651,7 @@ static const struct pad_config gpio_table[] = {
        /* ------- GPIO Group GPP_D ------- */
 
        /* GPP_D0 - SRCCLKREQ0# */
-       PAD_CFG_NF(GPP_D0, NONE, DEEP, NF1),
+       PAD_NC(GPP_D0, NONE),
        /* GPP_D1 - GPIO */
        PAD_CFG_GPI_TRIG_OWN(GPP_D1, NONE, PLTRST, OFF, ACPI),
        /* GPP_D2 - GPIO */```

**Comment by macpijan** [comment]
There is not `4.21` version of Dasharo. Is it binary released by Dasharo (the one based on 4.21 coreboot is not yet official released) or is it something you have built on your own? If so, please provide the exact build steps and source code revision.

**Comment by miczyg1** [comment]
Setting GPP_D0 to NC introduces a bug. GPP_D0 must be set to the native function of SRCCLKREQ0# for the ASPM to work.

ASPM is not enabled in the Dasharo builds in the first place and the version 4.21 indicates you have built from upstream.
